birdhouse/optional-components/test-cowbird-jupyter-access/test_cowbird_jupyter.py

```diff
@@ -0,0 +1,188 @@
+    if resp.status_code != 201:
+        raise ValueError(response_msg("\nCould not create test user [{}]".format(user_name), resp))
+    session.cookies = magpie_signin(user_name, password).cookies
+    return resp.json()
+
+# Make sure Cowbird is running before creating user
+resp = None
+for i in range(MAX_ATTEMPTS):
+    try:
+        resp = requests.get(COWBIRD_URL, verify=VERIFY_SSL)
+        assert resp.status_code == 200
+        print("Cowbird availability checked successfully.")
+        break
+    except Exception as exc:
+        print(f"Failed to connect to Cowbird [{exc}]. \nAttempting again ({i + 1})...")
+    time.sleep((i + 1) * TIMEOUT_DELAY)
+else:
+    raise ConnectionError("Failed to connect to Cowbird on url {}".format(COWBIRD_URL))
+
+# ------------------------------------------------------------------
+magpie_admin_session = requests.Session()
+magpie_admin_session.verify = VERIFY_SSL
+magpie_admin_session.headers = HEADERS
+magpie_admin_session.cookies = magpie_signin(TEST_MAGPIE_ADMIN_USERNAME, TEST_MAGPIE_ADMIN_PASSWORD).cookies
+
+test_user_session = requests.Session()
+test_user_session.verify = VERIFY_SSL
+test_user_session.headers = HEADERS
+
+# ------------------------------------------------------------------
+print("Creating test user on Magpie...")
+test_user_id = create_magpie_user(TEST_USER, TEST_PASSWORD, test_user_session)["user"]["user_id"]
+
+user_workspace_dir = f"{WORKSPACE_DIR}/{TEST_USER}"
+
+# Make sure cowbird had time to create user workspace before executing following operations
+for i in range(MAX_ATTEMPTS):
+    if os.path.exists(user_workspace_dir):
+        print(f"User workspace successfully found at path `{user_workspace_dir}`.")
+        break
+    print(f"Failed to find user workspace at path [{user_workspace_dir}]. Attempting again ({i + 1})...")
+    time.sleep((i + 1) * TIMEOUT_DELAY)
+else:
+    raise ConnectionError("Failed to create user workspace  to Cowbird on url {}".format(COWBIRD_URL))
+
+# ------------------------------------------------------------------
+# Add Geoserver shapefiles to the user workspace
+print("Copying shapefile data to user workspace...")
+GEOSERVER_TEST_DATA_DIR = "/geoserver-test-data"
+for filename in os.listdir(GEOSERVER_TEST_DATA_DIR):
+    file = Path(filename)
+    stem, ext = file.stem, file.suffix
+    # Make one read-only and one read-write copy
+    for target_path, perms in [(os.path.join(f"{user_workspace_dir}/shapefile_datastore", f"{stem}_ro{ext}"), 0o664),
+                               (os.path.join(f"{user_workspace_dir}/shapefile_datastore", f"{stem}_rw{ext}"), 0o666)]:
+        shutil.copy2(os.path.join(GEOSERVER_TEST_DATA_DIR, filename), target_path)
+        os.chmod(target_path, perms)
+
+# ------------------------------------------------------------------
+# Add wps-outputs
+print("Creating test WPS outputs data...")
+# Add public file
+public_wpsoutputs_filepath = f"{WPS_OUTPUTS_DIR}/weaver/public/test_public_file.txt"
```

Comment:
Now I see the source of the path from the other PR https://github.com/Ouranosinc/PAVICS-e2e-workflow-tests/pull/131#discussion_r1394735756. 

Please adjust it just to demonstrate what would be really expected, so it represents a real use case.
Example: 
```
/data/wps_outputs
└── weaver
    ├── 0fb1371d-48bb-4c59-9939-4ec6f02954b5.log
    ├── 0fb1371d-48bb-4c59-9939-4ec6f02954b5.xml
    ├── 63de2ff8-2ee5-40d0-a254-fdb493a35d77.log
    ├── 63de2ff8-2ee5-40d0-a254-fdb493a35d77.xml
    ├── 950776da-0d99-4607-a68a-c743493b56e5.log
    ├── 950776da-0d99-4607-a68a-c743493b56e5.xml
    ├── public
    │   ├── 63de2ff8-2ee5-40d0-a254-fdb493a35d77
    │   │   └── stdout.log
    │   ├── 3dc704a8-e1f6-4fe8-8bf0-153cf1e52c7a
    │   │   └── output
    │   │       └── nc_dump_jjEIOJ.txt
    │   ├── 3dc704a8-e1f6-4fe8-8bf0-153cf1e52c7a.log
    │   └── 3dc704a8-e1f6-4fe8-8bf0-153cf1e52c7a.xml
    └── users
        ├── 1
        │   ├── 0fb1371d-48bb-4c59-9939-4ec6f02954b5
        │   │   └── nc_dump_lko95z.txt
        │   └── 950776da-0d99-4607-a68a-c743493b56e5
        │       └── stdout.log
        └── 18
            ├── 075ed2a9-7b79-4444-99a2-840092e0a26b
            │   └── output
            │       └── nc_dump_s8zoWZ.txt
            ├── 075ed2a9-7b79-4444-99a2-840092e0a26b.log
            └── 075ed2a9-7b79-4444-99a2-840092e0a26b.xml
```